src/pudl/helpers.py

```diff
@@ -1798,3 +1798,36 @@ def scale_by_ownership(
         gens["fraction_owned"], axis="index"
     )
     return gens
+
+
+def get_dagster_execution_config(num_workers: int = 0):
+    """Get the dagster execution config for a given number of workers.
+
+    If num_workers is 0, then the dagster execution config will not include
+    any limits. With num_workesr set to 1, we will use in-process serial
+    executor, otherwise multi-process executor with maximum of num_workers
+    will be used.
+
+    Args:
+        num_workers: The number of workers to use for the dagster execution config.
+            If 0, then the dagster execution config will not include a multiprocess
+            executor.
+
+    Returns:
+        A dagster execution config.
+    """
+    if not num_workers:
+        return {}
+    if num_workers == 1:
+        return {
+            "execution": {
+                "config": {
+                    "in_process": {},
```

Comment:
Note that this behavior controls the execution when CLI tools are executed directly, and should not really change how tests are invoked. I'm not sure I fully understand the problem here, but my guess is that tests are currently not fully isolated and leak some materialization into shared `$DAGSTER_HOME` in which case, I agree that we should probably spin up temporary directory when running tests whenever we are not running with `--live-dbs`. With that flag, no ETL should really be run anyways, right?

Reply:
It seems to me that the two concerns (how we run CLIs and how we run tests) should be orthogonal concerns?

Reply:
It seemed to me like they overlap a bit because in this PR we're using the CLI to run the ETL for the tests rather than using the test fixture.

With the in-process executor the default IO Manager doesn't write anything to disk in `$DAGSTER_HOME`, it keeps the assets in memory (the assets that are written to the database still get written to the database, wherever the PUDL DB path says that DB is). And currently the tests inherit the user's `$DAGSTER_HOME` so if anything in the tests attempts to read from that directory, it's reading from the user's preexisting outputs rather than outputs that were generated by the tests, so a test that relies on an intermediate output that was written to disk can work when the user runs the tests locally (since they've got a `$DAGSTER_HOME` full of pickled dataframes) and then fail in the GitHub CI, where no intermediate dataframes have ever been written to disk, since the ETL has only ever been run in-process there.

So my guess is that what will happen now when the tests that use the CLI to build the DB are run locally is that they will overwrite the user's pickled dataframes, probably replacing `etl_full` outputs with `etl_fast` outputs.

Reply:
I think that the current plan was to use CLI for ETL for integration tests only in the context of CI infrastructure and there it should be using multiprocess with no limits. This change should not affect the default behavior, but if people want to run single threaded, they can (e.g. for perf analysis).

Do we also want to make use of separating ETL for the purpose of locally run integration tests? If so, we probably need to do some additional work.